The STM32MP1 series chips can't do 24-bit DMA transfers and so the 
SNDRV_PCM_FMTBIT_S24_LE format gets ignored by the hardware-level 
driver resulting in an error about "No matching formats". The 
PCM3060 chips seems to work fine with S32_LE format, although I'm
not entirely sure why.

diff --git a/sound/soc/codecs/pcm3060.c b/sound/soc/codecs/pcm3060.c
index 586ec8c..7fe1262 100644
--- a/sound/soc/codecs/pcm3060.c
+++ b/sound/soc/codecs/pcm3060.c
@@ -177,6 +177,8 @@ static const struct snd_soc_dai_ops pcm3060_dai_ops = {
 #define PCM3060_DAI_RATES_DAC	(PCM3060_DAI_RATES_ADC | \
 				 SNDRV_PCM_RATE_176400 | SNDRV_PCM_RATE_192000)
 
+#define PCM3060_FORMATS (SNDRV_PCM_FMTBIT_S32_LE)
+
 static struct snd_soc_dai_driver pcm3060_dai[] = {
 	{
 		.name = "pcm3060-dac",
@@ -186,7 +188,7 @@ static struct snd_soc_dai_driver pcm3060_dai[] = {
 			.channels_min = 2,
 			.channels_max = 2,
 			.rates = PCM3060_DAI_RATES_DAC,
-			.formats = SNDRV_PCM_FMTBIT_S24_LE,
+			.formats = PCM3060_FORMATS,
 		},
 		.ops = &pcm3060_dai_ops,
 	},
@@ -198,7 +200,7 @@ static struct snd_soc_dai_driver pcm3060_dai[] = {
 			.channels_min = 2,
 			.channels_max = 2,
 			.rates = PCM3060_DAI_RATES_ADC,
-			.formats = SNDRV_PCM_FMTBIT_S24_LE,
+			.formats = PCM3060_FORMATS,
 		},
 		.ops = &pcm3060_dai_ops,
 	},
